### PR TITLE
[FIX] Markdown Rules for Multiple Newlines in Messages

### LINKED ISSUE #1288: [BUG][chat window] Double New line characters are collapsed to a single new line in RC livechat chat windows.
We would like to be able to send messages from DF.agent that were formatted with new line characters.  Currently when those are displayed in the widget the new line character is stripped and the lines are collapsed. 

Same as described in this issue:  https://github.com/RocketChat/feature-requests/issues/385

### PR DESCRIPTION
**Pull request by AlexanderKanakis**
Fixes: https://github.com/WideChat/Rocket.Chat/issues/1288

### CODE REVIEW

**Comment by ear-dev** [comment]
@AlexanderKanakis can you fix the merge conflict please?  thanks.

**Comment by AlexanderKanakis** [comment]
Merge conflict fixed.

**Review by Shailesh351 — APPROVED** [review]
@AlexanderKanakis This looks good, working as expected in my local. Can you share any reference/link for this solution if possible?

**Comment by AlexanderKanakis** [comment]
> @AlexanderKanakis This looks good, working as expected in my local. Can you share any reference/link for this solution if possible?

https://markdown-it.github.io/markdown-it/#Renderer
https://github.com/markdown-it/markdown-it/issues/617
https://github.com/markdown-it/markdown-it/blob/master/lib/rules_block/paragraph.js

**Comment by ear-dev** [comment]
NOTE: Testing this with the most recent version of develop_pwa branch and the new line fix does not work.  Alexander verified that he is seeing the same thing and will fix.

**Comment by ear-dev** [comment]
thanks @AlexanderKanakis this looks good now.  @Shailesh351 can you take a look at the latest commits to this PR and approve again please?  Thanks.

File: src/components/Messages/MessageText/markdown.js [hunk 1]
```diff
@@ -29,6 +29,58 @@ md.renderer.rules.link_open = (tokens, idx, options, env, self) => {
...
+				if (diff > 0) {
+					result = '<br>'.repeat(diff);
+				}
+			}
+		}
+	}
+	return result + defaultRender(tokens, idx, options, env, self);
+};
+
+md.renderer.rules.list_item_open = function(tokens, idx, options, env, self) {
```
Comment (Shailesh351) [inline src/components/Messages/MessageText/markdown.js:62]: PR working as expected on my local. @AlexanderKanakis Is this implementation our own or from any reference? I'm sorry, It's new for me so just trying to understand what's going on. Why `list_item_open`? If the rules for both are almost the same(just the difference with some indices),  is creating common fn a good idea?

Comment (AlexanderKanakis) [reply src/components/Messages/MessageText/markdown.js:62]: @Shailesh351 `list_item_open` contains items in a list. A `list_item_open` item contains a `paragraph_open` so we could edit only one ruleset, but checking both makes it easier to handle newline insertion in-between list items and before the list.

**Review by Shailesh351 — APPROVED** [review]
LGTM!

Screenshot 2022-09-13 at 11 07 07 PM

**Pull request merged**